### PR TITLE
CRM-19836 Enable activity create from replies to CiviMail.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
This relies on a parameter being added to the api call:

civicrm_api3('Job', 'fetch_bounces', array('is_create_activities' => TRUE));

---

 * [CRM-19836: Allow mail bounce processing to create activities for inbound email](https://issues.civicrm.org/jira/browse/CRM-19836)

### CODE REVIEW

File: api/v3/Job.php [hunk 1]
```diff
@@ -370,6 +370,19 @@ function civicrm_api3_job_fetch_bounces($params) {
 }
 
 /**
+ * Metadata for bounce function.
+ *
+ * @param array $params
+ */
+function _civicrm_api3_job_fetch_bounces_spec(&$params) {
+  $params['is_create_activities'] = array(
```
Comment (colemanw) [inline api/v3/Job.php]: I like the grammar of "is create activities" but what about "can I haz activities?"

File: CRM/Utils/Mail/EmailProcessor.php [hunk 2]
```diff
@@ -48,14 +48,14 @@ class CRM_Utils_Mail_EmailProcessor {
    *   Always returns true (for the api). at a later stage we should
    *   fix this to return true on success / false on failure etc.
    */
-  public static function processBounces() {
+  public static function processBounces($is_create_activities) {
```
Comment (colemanw) [inline CRM/Utils/Mail/EmailProcessor.php]: this variable needs to go in the docblock

File: api/v3/Job.php [hunk 3]
```diff
@@ -358,7 +358,7 @@ function civicrm_api3_job_fetch_bounces($params) {
   if (!$lock->isAcquired()) {
     return civicrm_api3_create_error('Could not acquire lock, another EmailProcessor process is running');
   }
-  if (!CRM_Utils_Mail_EmailProcessor::processBounces()) {
+  if (!CRM_Utils_Mail_EmailProcessor::processBounces($params['is_create_activities'])) {
```
Comment (colemanw) [inline api/v3/Job.php]: Are you sure this array key will always be set?

File: api/v3/Job.php (hunk 1 above)
Comment (eileenmcnaughton) [inline api/v3/Job.php]: no you may not! but if you look at the line below this comment, that is why I'm sure the var will always be set. 

I'll fix the data block

**Comment by eileenmcnaughton** [comment]
@colemanw I fixed the data block. I did a bit of clean up on the fantasy return value. processBounces is only called from the api, which handles the possibility a function that never returns false might someday feel like it. I put it out of it's misery

**Comment by colemanw** [comment]
I think the `can_haz_activities` param might better belong in the `options` array rather than `params` but whatever, that's a mess we'll clean up in api4.

**Pull request merged**